src/typesOfSchema.ts

```diff
@@ -35,6 +35,9 @@ const matchers: Record<SchemaType, (schema: JSONSchema) => boolean> = {
   ALL_OF(schema) {
     return 'allOf' in schema
   },
+  NEVER(schema) {
+    return (schema as any) === false
```

Comment:
Can you avoid this type assertion, and instead ensure upstream that the type for `schema` declares that it can be a `JSONSchema | boolean`?

(If you're not comfortable doing that, I can do it as a follow-up after we merge this PR.)